### PR TITLE
Consider evaluating association in serializer context

### PR DESCRIPTION
**Pull request by bf4**
For discussion:

Consider evaluating association in serializer context

That way, associations are really just anything that
can be conditionally included.  They no longer
have to actually be methods on the object or serializer.

e.g.

```diff
has_many :comments do
- last(1)
+ Comment.active.for_serialization(object).last(1)
end
```
ref: https://github.com/rails-api/active_model_serializers/pull/1356

### CODE REVIEW

**Comment by beauby** [comment]
This is a must IMO. The current behavior is confusing.

**Comment by beauby** [comment]
We should merge this. How should we go about warning people migrating from previous versions?

**Comment by bf4** [comment]
Changelog? Update doc?

B mobile phone

**Comment by bf4** [comment]
@beauby Updated CHANGELOG https://github.com/rails-api/active_model_serializers/blob/d7de53ce30aeb9a6600614e6eabff10d8b60031c/CHANGELOG.md

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,7 +16,22 @@ Breaking changes:
 
 Features:
 
-- [#1336](https://github.com/rails-api/active_model_serializers/pull/1336) Added support for Grape >= 0.13, < 1.0
+- [#1378](https://github.com/rails-api/active_model_serializers/pull/1378) Change association blocks
+  to be evaluated in *serializer* scope, rather than *association* scope. (@bf4)
+    * Syntax changes from e.g.
+      `has_many :titles do customers.pluck(:title) end` (in #1356) to
+      `has_many :titles do object.customers.pluck(:title) end`
+- [#1356](https://github.com/rails-api/active_model_serializers/pull/1356) Add inline syntax for
```
Comment (beauby) [inline CHANGELOG.md]: Does this belong here?

Comment (bf4) [inline CHANGELOG.md]: why wouldn't it?

Comment (beauby) [inline CHANGELOG.md]: Because this PR is #1378 and not #1356

Comment (bf4) [inline CHANGELOG.md]: Oh... They're just missing from changelog.  How would you prefer i include them?

B mobile phone

Comment (bf4) [inline CHANGELOG.md]: I added it to my local master and merged locally so it's two commits, but not part of this PR merge commit
